### DISCUSSION TITLE
How to add enchantments to item?

### DISCUSSION THREAD

**Discussion by SirEnder125** [FabricMC/fabric-api#2125] [Mod Dev Support]
Hey. I want to add an enchantment to my item, so that every ItemStack will have that enchantment. How can I do this? (To be exact, I don't just want hasGlint() to return true, but an actual effect.)

**Comment by frqnny** [comment]
Look at vanilla’s EnchantmentHelper class

**Reply by SirEnder125** [reply]
Looks interesting, I noticed an enchant() method, but where should I call that? 🤔

**Reply by frqnny** [reply]
Anywhere you want to enchant in code.


An important note, though, is to ensure you handle networking correctly.
You cannot use it on the client, you might need to look into packets if you run into issues. Otherwise, have fun.

**Reply by SirEnder125** [reply]
Hmm, okay. Where in the Item class can you make it where an ItemStack is enchanted on creation? Also, what issues might I run in to?

**Reply by frqnny** [reply]
idk but might want to see if mixin might be able to accomplish it if you can’t find a way.

**Reply by SirEnder125** [reply]
Hmm. I first tried enchanting the return variable of Item.getDefaultStack(), but the code wasn't called when I got the item from creative menu.

**Reply by warjort** [reply]
If you want different versions of your item in the creative mode tabs, you need to implement ````Item.appendStacks()````.
e.g. add different ItemStacks to the ItemGroup, one with an enchantment and one without.
Take a look at EnchantedBookItem for an example, which adds an ItemStack for each known enchantment.

**Reply by warjort** [reply]
As for creating an ItemStack with a default enchantment, I don't think Minecraft has a callback for that?
You either have to create the original stack yourself or you can do something in ````Item.onCraft()````.

To trap every ItemStack creation, you would need to mixin the tail of the constructors.
But you will get a lot more calls than you probably expect. e.g. copying, transfer over the network, splitting item stacks, etc.
